modules/dnn/src/opencl/region.cl

```diff
@@ -44,13 +44,18 @@
 __kernel void logistic_activ(const int count,
                              __global const Dtype* src,
                              const int cell_size,
-                             __global Dtype* dst)
+                             __global Dtype* dst,
+                             const int batch_size,
+                             const int sample_size)
 {
-    for (int i = get_global_id(0); i < count; i += get_global_size(0))
+    for (int b = get_global_id(0); b < batch_size; b += get_global_size(0))
```

Comment:
It looks like we can remove this `batch_size` as we did for CPU implementation but pass `count` already multiplied by this value.